platform/ethereum/blockbook/transaction.go

```diff
@@ -0,0 +1,146 @@
+package blockbook
+
+import (
+	"strings"
+
+	"github.com/trustwallet/blockatlas/coin"
+	"github.com/trustwallet/blockatlas/pkg/blockatlas"
+)
+
+func (c *Client) GetTransactions(address string, coinIndex uint) (blockatlas.TxPage, error) {
+	page, err := c.GetTxs(address)
+	if err != nil {
+		return nil, err
+	}
+	return NormalizePage(page, address, "", coinIndex), nil
+}
+
+func (c *Client) GetTokenTxs(address, token string, coinIndex uint) (blockatlas.TxPage, error) {
+	page, err := c.GetTxsWithContract(address, token)
+	if err != nil {
+		return nil, err
+	}
+	return NormalizePage(page, address, token, coinIndex), nil
+}
+
+func NormalizePage(srcPage *Page, address, token string, coinIndex uint) blockatlas.TxPage {
+	var txs []blockatlas.Tx
+	for _, srcTx := range srcPage.Transactions {
+		tx := normalizeTxWithAddress(&srcTx, address, token, coinIndex)
+		txs = append(txs, tx)
+	}
+	return txs
+}
+
+func normalizeTx(srcTx *Transaction, coinIndex uint) blockatlas.Tx {
+	status, errReason := getStatus(srcTx.EthereumSpecific)
+	from := getFrom(srcTx)
+	to := getTo(srcTx)
+
+	return blockatlas.Tx{
+		ID:       srcTx.TxID,
+		Coin:     coinIndex,
+		From:     from,
+		To:       to,
+		Fee:      blockatlas.Amount(srcTx.Fees),
+		Date:     srcTx.BlockTime,
+		Block:    srcTx.BlockHeight,
+		Status:   status,
+		Error:    errReason,
+		Sequence: srcTx.EthereumSpecific.Nonce,
+	}
+}
+
+func normalizeTxWithAddress(srcTx *Transaction, address, token string, coinIndex uint) blockatlas.Tx {
+	normalized := normalizeTx(srcTx, coinIndex)
+	normalized.Direction = getDirection(address, normalized.From, normalized.To)
+	fillMeta(&normalized, srcTx, address, token, coinIndex)
+	return normalized
+}
+
+func getStatus(specific *EthereumSpecific) (blockatlas.Status, string) {
```

Comment:
can you add extension to `EthereumSpecific` instead as `EthereumSpecific.GetStatus` ?

Reply:
done